File: myoro_matchup_api/src/main/java/com/myoro/myoro_matchup_api/dto/InvitationResponseDto.java
package com.myoro.myoro_matchup_api.dto;

import com.myoro.myoro_matchup_api.enums.InvitationStatusEnum;

import java.time.LocalDateTime;

/** Invitation response DTO. */
public class InvitationResponseDto {
  /** Invitation ID. */
  private Long id;

  /** Game being invited to. */
  private InvitationGameResponseDto game;

  /** Name of the user who sent the invitation. */
  private String inviterName;

  /** Status of the invitation. */
  private InvitationStatusEnum status;

  /** When the invitation was created. */
  private LocalDateTime createdAt;

  /** When the invitation expires. */
  private LocalDateTime expiresAt;

  /** When the invitee responded. */
  private LocalDateTime respondedAt;

  /** Optional message from the inviter. */
  private String message;

  /** Default constructor. */
  public InvitationResponseDto() {
  }

  /**
   * Getter for id
   * 
   * @return the invitation ID
   */
  public Long getId() {
    return id;
  }

  /**
   * Getter for game
   * 
   * @return the game being invited to
   */
  public InvitationGameResponseDto getGame() {
    return game;
  }

  /**
   * Getter for inviterName
   * 
   * @return the name of the user who sent the invitation
   */
  public String getInviterName() {
    return inviterName;
  }

  /**
   * Getter for status
   * 
   * @return the status of the invitation
   */
  public InvitationStatusEnum getStatus() {
    return status;
  }

  /**
   * Getter for createdAt
   * 
   * @return when the invitation was created
   */
  public LocalDateTime getCreatedAt() {
    return createdAt;
  }

  /**
   * Getter for expiresAt
   * 
   * @return when the invitation expires
   */
  public LocalDateTime getExpiresAt() {
    return expiresAt;
  }

  /**
   * Getter for respondedAt
   * 
   * @return when the invitee responded
   */
  public LocalDateTime getRespondedAt() {
    return respondedAt;
  }

  /**
   * Getter for message
   * 
   * @return the optional message from the inviter
   */
  public String getMessage() {
    return message;
  }

  /**
   * Setter for game
   * 
   * @param game the game being invited to
   */
  public void setGame(InvitationGameResponseDto game) {
    this.game = game;
  }

  /**
   * Setter for id
   * 
   * @param id the invitation ID
   */
  public void setId(Long id) {
    this.id = id;
  }

  /**
   * Setter for inviterName
   * 
   * @param inviterName the name of the user who sent the invitation
   */
  public void setInviterName(String inviterName) {
    this.inviterName = inviterName;
  }

  /**
   * Setter for status
   * 
   * @param status the status of the invitation
   */
  public void setStatus(InvitationStatusEnum status) {
    this.status = status;
  }

  /**
   * Setter for createdAt
   * 
   * @param createdAt when the invitation was created
   */
  public void setCreatedAt(LocalDateTime createdAt) {
    this.createdAt = createdAt;
  }

  /**
   * Setter for expiresAt
   * 
   * @param expiresAt when the invitation expires
   */
  public void setExpiresAt(LocalDateTime expiresAt) {
    this.expiresAt = expiresAt;
  }

  /**
   * Setter for respondedAt
   * 
   * @param respondedAt when the invitee responded
   */
  public void setRespondedAt(LocalDateTime respondedAt) {
    this.respondedAt = respondedAt;
  }

  /**
   * Setter for message
   * 
   * @param message the optional message from the inviter
   */
  public void setMessage(String message) {
    this.message = message;
  }
}

File: myoro_matchup/lib/module/invitation/dto/invitation_response_dto.dart
import 'package:faker/faker.dart';
import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:myoro_flutter_annotations/myoro_flutter_annotations.dart';
import 'package:myoro_flutter_library/myoro_flutter_library.dart';
import 'package:myoro_matchup/myoro_matchup.dart';

part 'invitation_response_dto.g.dart';

/// Invitation response DTO.
@immutable
@myoroModel
@JsonSerializable(createToJson: false)
final class InvitationResponseDto with _$InvitationResponseDtoMixin {
  /// Default constructor.
  const InvitationResponseDto({
    required this.id,
    required this.game,
    required this.inviterName,
    required this.status,
    required this.createdAt,
    required this.expiresAt,
    required this.respondedAt,
    required this.message,
  });

  /// From JSON.
  factory InvitationResponseDto.fromJson(Map<String, dynamic> json) {
    return _$InvitationResponseDtoFromJson(json);
  }

  /// Fake constructor.
  factory InvitationResponseDto.fake() {
    final createdAt = myoroFake<DateTime>();
    final expiresAt = createdAt.add(const Duration(days: 7));
    final respondedAt = createdAt.add(const Duration(days: 1));

    return InvitationResponseDto(
      id: faker.randomGenerator.integer(100000, min: 1),
      game: InvitationGameResponseDto.fake(),
      inviterName: faker.person.name(),
      status: InvitationStatusEnum.fake(),
      createdAt: createdAt,
      expiresAt: expiresAt,
      respondedAt: respondedAt,
      message: faker.randomGenerator.boolean() ? faker.lorem.sentence() : kMyoroEmptyString,
    );
  }

  /// Game being invited to.
  final InvitationGameResponseDto game;

  /// Invitation identifier.
  final int id;

  /// Name of the user who sent the invitation.
  final String inviterName;

  /// Status of the invitation.
  final InvitationStatusEnum status;

  /// When the invitation was created.
  final DateTime createdAt;

  /// When the invitation expires.
  final DateTime expiresAt;

  /// When the invitee responded.
  final DateTime? respondedAt;

  /// Optional message from the inviter.
  @JsonKey(defaultValue: kMyoroEmptyString)
  final String message;
}
